platforms/core-execution/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonClientsManager.java

```diff
@@ -157,22 +170,23 @@ private void stopWorkers(List<WorkerDaemonClient> clientsToStop) {
         if (clientsToStop.size() > 0) {
             int clientCount = clientsToStop.size();
             LOGGER.debug("Stopping {} worker daemon(s).", clientCount);
-            List<Exception> failures = new ArrayList<>();
+            int failureCount = 0;
             for (WorkerDaemonClient client : clientsToStop) {
                 try {
-                    client.stop();
+                    if (client.isFailed()) {
+                        emitUnexpectedWorkerFailureWarning(client);
+                    } else {
+                        client.stop();
+                    }
                 } catch (Exception e) {
-                    failures.add(e);
+                    failureCount++;
+                    LOGGER.warn("Failed to stop worker daemon '" + client.getDisplayName() + "'", e);
                 }
             }
             idleClients.removeAll(clientsToStop);
             allClients.removeAll(clientsToStop);
-            if (!failures.isEmpty()) {
-                if (failures.size() == 1) {
-                    throw UncheckedException.throwAsUncheckedException(failures.get(0));
-                } else {
-                    throw new DefaultMultiCauseException("Not all worker daemon(s) could be stopped.", failures);
-                }
+            if (failureCount > 0) {
```

Comment:
Is this info log even necessary anymore? We only print it if we have a failure stopping a worker, and in which case we print a warning for each individually

Reply:
I think it can be interesting to know how many worker daemons were stopped (failed or not) while troubleshooting, but I could be convinced that this is not interesting at info level if you feel strongly about it.  I'm inclined to leave this in for now since it's only a single message and we're not adding something super chatty to the info log.

Reply:
No worries you can leave it in if you think its useful